File: gen-README.py
def processPicture(lines: list):
    return f"<a {lines[0]} target=\"_blank\"><img {lines[1]} {lines[2]} {lines[3]}/></a>\n"

def processBlock(block: list):
    out = ""
    out += processPicture(block[0])
    if(len(block) > 1):
        out += "&nbsp;&nbsp;&nbsp; &nbsp; &nbsp;\n"
        for fw in block[1]:
            out += processPicture(fw)
    out +='\n'
    return out


open("README.md", "w").write(open("template-intro.md", "r").read() + ''.join([processBlock(l) for l in [[[k for k in j.split('\n') if k != ""] if c == 0 else [k.split('\n') for k in j.split("-"*32 + '\n') if k != ""] for c, j in enumerate(i.split("-" * 32 + '\n' + "-" * 32 + '\n'))] for i in open("frameworks", "r").read().split("\n\n\n")]]) + open("template-outro.md", "r").read())

File: gen-README.hs
module Generator where

compileA :: String -> String -> String
compileA href cont = "<a " ++ href ++ " target=\"_blank\">" ++ cont ++ "</a>"

compileImg :: String -> String -> String -> String
compileImg src alt height = "<img " ++ src ++ " " ++ alt ++ " " ++ height ++ "/>"

compileToTag :: [String] -> String
compileToTag l = compileA (head l) $ compileImg (l !! 1) (l !! 2) (l !! 3)

compileAll :: ([String], [[String]]) -> String
compileAll (h, b) = unlines $ [compileToTag h, "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"] ++ map compileToTag b

split :: Eq a => a -> [a] -> [[a]]
split _ [] = []
split e l = f : split e (drop 1 ls)
  where
    (f, ls) = span (/= e) l

sections :: String -> [[String]]
sections = filter (not . null) . split "" . lines

headAndBody :: [String] -> ([String], [[String]])
headAndBody all = (heading, body)
  where
    s = split "--------------------------------" all
    heading = head s
    body = filter (not . null) $ tail s

main = do
  intro <- readFile "./template-intro.md"
  frameworks <- readFile "./frameworks"
  outro <- readFile "./template-outro.md"
  writeFile "README.md" $ unlines $ [intro] ++ map (compileAll . headAndBody) (sections frameworks) ++ [outro]
